app/controllers/organizations/adoption_application_reviews_controller.rb

```diff
@@ -1,9 +1,17 @@
 class Organizations::AdoptionApplicationReviewsController < Organizations::BaseController
-  before_action :active_staff
+  before_action :set_adopter_application, only: %i[edit update]
+
   layout "dashboard"
 
   def index
-    @q = Pet.org_pets_with_apps(current_user.staff_account.organization_id).ransack(params[:q])
+    authorize! AdopterApplication,
+      context: {organization: current_user.organization},
+      with: Organizations::AdopterApplicationReviewPolicy
```

Comment:
Do we need the `with` here? It looks like it would be handled correctly -> https://github.com/palkan/action_policy/blob/master/docs/namespaces.md

Reply:
We would need `with` here, at least with how I had the policy written. I thought we could use namespace here as well. However, the lookup pattern is looking at the model of the object passed which does not match the controllers name, so if we do:
```ruby
    authorize! AdopterApplication,
      context: {organization: current_user.organization},
      namespace: Organizations
```
It would look for the policy `Organizations::AdopterApplicationPolicy`, not `Organizations::AdoptionApplicationReviewPolicy`. 

However, you got me thinking, we should just rename the policy to match the model in the first place, and then we can also use the namespace lookup. I've implemented that in b90e8319dd392c227b12d854e4a35fd5afef078a. This is also better since the policy actually didn't match the controller anyway because the controller name uses "adoption" not "adopter".

Reply:
**BUT**, this brings up a different point for the authorized_scope usage here. I was forcing the scope to use the `Organizations::AdopterApplicationReviewPolicy` for getting the `relation_scope` here before. If I remove that `with`, it actually is going to go for the `Organizations::PetPolicy` to check the `relation_scope` there. In this particular case, both policies are using the same defaults from `ApplicationPolicy`, so it does not make an immediate difference.

It makes me almost wonder if @q is grabbing the wrong relation here. This is an `index` for `AdoptionApplicationReviews` but it's not indexing applications, it's indexing pets (with the apps merged with pets). But AdopterApplications already have a pet reference so couldn't we just use ransack on the adopter applications themselves? 

I've made a commit to remove the `with` from the scope and change the controller test to check that the controller scope is grabbing from `Organizations::PetPolicy`. c47110f004294a11a77694b485bf0db990e49a91

Not sure how I feel about it. I'll sleep on it and would love to here your thoughts @nsiwnf

Reply:
😵‍💫  Yeah, i see how this page indexing is weird - and yeah, seems like we should be able to group by pet through adoption applications - that might be more within the integrity of this controller., but honestly, I think it's fine as it is if the end result is what we expect! Also, I totally missed the "reviews" part of the policy 🤦. The rename to `AdopterApplicationPolicy` makes more sense to me tho!

Reply:
Yeah, I think I agree with you to leave the ransack search as is. I looked at a bit, and I can see why it was done this way. The page is using a partial from pets to show each pet's associated applications together. Changing it to index the applications, we would ultimately end up mapping the applications into pets anyway to keep using the shared partial.